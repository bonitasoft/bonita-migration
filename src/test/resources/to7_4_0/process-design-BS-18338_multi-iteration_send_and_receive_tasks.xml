<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<def:processDefinition id="0" name="IteratorCaller" version="1.1" xmlns:def="http://www.bonitasoft.org/ns/process/client/7.2">
    <description></description>
    <displayDescription></displayDescription>
    <parameters/>
    <actors>
        <actor name="Employee actor" initiator="true">
            <description>This is an example of actor that is mapped to any ACME users</description>
        </actor>
    </actors>
    <actorInitiator>Employee actor</actorInitiator>
    <flowElements id="7282544943062835405">
        <userTask id="4744803437825369503" name="Step1" actorName="Employee actor" expectedDuration="3600000" priority="NORMAL">
            <incomingTransition>7728963729586879939</incomingTransition>
            <outgoingTransition>6167313851637512958</outgoingTransition>
            <dataDefinitions/>
            <BusinessDataDefinitions/>
            <operations/>
            <boundaryEvents/>
            <contract>
                <inputDefinitions/>
                <constraints/>
            </contract>
            <context/>
        </userTask>
        <sendTask id="8657294294082783805" name="Step2">
            <incomingTransition>6167313851637512958</incomingTransition>
            <outgoingTransition>5236469411320165058</outgoingTransition>
            <dataDefinitions>
                <dataDefinition id="5101903968302105694" name="multiInstanceIterator" transient="false" className="java.lang.Integer"/>
            </dataDefinitions>
            <BusinessDataDefinitions/>
            <operations/>
            <multiInstanceLoopCharacteristics isSequential="false" loopDataInputRef="iter" dataInputItemRef="multiInstanceIterator"/>
            <boundaryEvents/>
            <throwMessageEventTrigger name="SendMessage">
                <targetProcess id="7015075233745235442" name="IteratorCallee" expressionType="TYPE_CONSTANT" returnType="java.lang.String" interpreter="">
                    <content>IteratorCallee</content>
                </targetProcess>
                <targetFlowNode id="4950270138380226155" name="Start2" expressionType="TYPE_CONSTANT" returnType="java.lang.String" interpreter="">
                    <content>Start2</content>
                </targetFlowNode>
            </throwMessageEventTrigger>
        </sendTask>
        <receiveTask id="6333978609595215485" name="Step3">
            <outgoingTransition>8443638224878586673</outgoingTransition>
            <dataDefinitions>
                <dataDefinition id="6878173575407106009" name="multiInstanceIterator" transient="false" className="java.lang.Integer"/>
            </dataDefinitions>
            <BusinessDataDefinitions/>
            <operations/>
            <multiInstanceLoopCharacteristics isSequential="false" loopDataInputRef="iter" dataInputItemRef="multiInstanceIterator"/>
            <boundaryEvents/>
            <catchMessageEventTrigger name="Notify Caller"/>
        </receiveTask>
        <transitions>
            <transition id="6167313851637512958" name="4744803437825369503_-&gt;_8657294294082783805" source="4744803437825369503" target="8657294294082783805"/>
            <transition id="7728963729586879939" name="9037969053042913707_-&gt;_4744803437825369503" source="9037969053042913707" target="4744803437825369503"/>
            <transition id="8443638224878586673" name="6333978609595215485_-&gt;_6369021331598608872" source="6333978609595215485" target="6369021331598608872"/>
            <transition id="5236469411320165058" name="8657294294082783805_-&gt;_6369021331598608872" source="8657294294082783805" target="6369021331598608872"/>
        </transitions>
        <startEvent id="9037969053042913707" name="Start1" interrupting="true">
            <outgoingTransition>7728963729586879939</outgoingTransition>
        </startEvent>
        <endEvent id="6369021331598608872" name="End1">
            <incomingTransition>5236469411320165058</incomingTransition>
            <incomingTransition>8443638224878586673</incomingTransition>
        </endEvent>
        <dataDefinitions>
            <dataDefinition id="6113089997408229152" name="iter" transient="false" className="java.util.Collection">
                <defaultValue id="6258696193085100433" name="iter" expressionType="TYPE_READ_ONLY_SCRIPT" returnType="java.util.List" interpreter="GROOVY">
                    <content>return [1, 2, 3, 4]</content>
                </defaultValue>
            </dataDefinition>
        </dataDefinitions>
        <businessDataDefinitions/>
        <documentDefinitions/>
        <documentListDefinitions/>
        <connectors/>
        <elementFinder/>
    </flowElements>
    <stringIndexes>
        <stringIndex index="1"/>
        <stringIndex index="2"/>
        <stringIndex index="3"/>
        <stringIndex index="4"/>
        <stringIndex index="5"/>
    </stringIndexes>
    <contract>
        <inputDefinitions/>
        <constraints/>
    </contract>
    <context/>
</def:processDefinition>
