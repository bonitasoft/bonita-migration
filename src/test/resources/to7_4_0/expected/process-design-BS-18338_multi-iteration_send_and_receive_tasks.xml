<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<def:processDefinition xmlns:def="http://www.bonitasoft.org/ns/process/client/7.4" id="_0" name="IteratorCaller" version="1.1">
    <description/>
    <displayDescription/>
    <parameters/>
    <actors>
        <actor initiator="true" id="_65537" name="Employee actor">
            <description>This is an example of actor that is mapped to any ACME users</description>
        </actor>
    </actors>
    <actorInitiator>_65537</actorInitiator>
    <flowElements id="_7282544943062835405">
        <receiveTask id="_6333978609595215485" name="Step3">
            <outgoingTransition>_8443638224878586673</outgoingTransition>
            <dataDefinitions>
                <dataDefinition id="_6878173575407106009" name="multiInstanceIterator" transient="false" className="java.lang.Integer"/>
            </dataDefinitions>
            <businessDataDefinitions/>
            <operations/>
            <multiInstanceLoopCharacteristics isSequential="false" loopDataInputRef="iter" dataInputItemRef="multiInstanceIterator"/>
            <boundaryEvents/>
            <catchMessageEventTrigger name="Notify Caller"/>
        </receiveTask>
        <sendTask id="_8657294294082783805" name="Step2">
            <incomingTransition>_6167313851637512958</incomingTransition>
            <outgoingTransition>_5236469411320165058</outgoingTransition>
            <dataDefinitions>
                <dataDefinition id="_5101903968302105694" name="multiInstanceIterator" transient="false" className="java.lang.Integer"/>
            </dataDefinitions>
            <businessDataDefinitions/>
            <operations/>
            <multiInstanceLoopCharacteristics isSequential="false" loopDataInputRef="iter" dataInputItemRef="multiInstanceIterator"/>
            <boundaryEvents/>
            <throwMessageEventTrigger name="SendMessage">
                <targetProcess id="_65597" name="IteratorCallee" expressionType="TYPE_CONSTANT" returnType="java.lang.String" interpreter="">
                    <content>IteratorCallee</content>
                </targetProcess>
                <targetFlowNode id="_65605" name="Start2" expressionType="TYPE_CONSTANT" returnType="java.lang.String" interpreter="">
                    <content>Start2</content>
                </targetFlowNode>
            </throwMessageEventTrigger>
        </sendTask>
        <userTask id="_4744803437825369503" name="Step1" actorName="Employee actor" priority="NORMAL">
            <incomingTransition>_7728963729586879939</incomingTransition>
            <outgoingTransition>_6167313851637512958</outgoingTransition>
            <dataDefinitions/>
            <businessDataDefinitions/>
            <operations/>
            <boundaryEvents/>
            <contract>
                <inputDefinitions/>
                <constraints/>
            </contract>
            <context/>
            <expectedDuration expressionType="TYPE_CONSTANT" id="_65560" returnType="java.lang.Long" name="expectedDuration expression">
                <content>3600000</content>
            </expectedDuration>
        </userTask>
        <transitions>
            <transition id="_6167313851637512958" name="4744803437825369503_-&gt;_8657294294082783805" source="_4744803437825369503" target="_8657294294082783805"/>
            <transition id="_7728963729586879939" name="9037969053042913707_-&gt;_4744803437825369503" source="_9037969053042913707" target="_4744803437825369503"/>
            <transition id="_8443638224878586673" name="6333978609595215485_-&gt;_6369021331598608872" source="_6333978609595215485" target="_6369021331598608872"/>
            <transition id="_5236469411320165058" name="8657294294082783805_-&gt;_6369021331598608872" source="_8657294294082783805" target="_6369021331598608872"/>
        </transitions>
        <startEvent id="_9037969053042913707" name="Start1" interrupting="true">
            <outgoingTransition>_7728963729586879939</outgoingTransition>
        </startEvent>
        <endEvent id="_6369021331598608872" name="End1">
            <incomingTransition>_5236469411320165058</incomingTransition>
            <incomingTransition>_8443638224878586673</incomingTransition>
        </endEvent>
        <dataDefinitions>
            <dataDefinition id="_6113089997408229152" name="iter" transient="false" className="java.util.Collection">
                <defaultValue id="_65673" name="iter" expressionType="TYPE_READ_ONLY_SCRIPT" returnType="java.util.List" interpreter="GROOVY">
                    <content>return [1, 2, 3, 4]</content>
                </defaultValue>
            </dataDefinition>
        </dataDefinitions>
        <businessDataDefinitions/>
        <documentDefinitions/>
        <documentListDefinitions/>
        <connectors/>
        <elementFinder/>
    </flowElements>
    <stringIndexes>
        <stringIndex index="1"/>
        <stringIndex index="2"/>
        <stringIndex index="3"/>
        <stringIndex index="4"/>
        <stringIndex index="5"/>
    </stringIndexes>
    <contract>
        <inputDefinitions/>
        <constraints/>
    </contract>
    <context/>
</def:processDefinition>